database/DatabaseExampleSwift/SignInViewController.swift

```diff
@@ -36,97 +36,150 @@ class SignInViewController: UIViewController, UITextFieldDelegate {
   }
 
   @IBAction func didTapEmailLogin(_ sender: AnyObject) {
-    if let email = self.emailField.text, let password = self.passwordField.text {
-      showSpinner({
-        FIRAuth.auth()?.signIn(withEmail: email, password: password) { (user, error) in
-          self.hideSpinner({
-            if let error = error {
-              self.showMessagePrompt(error.localizedDescription)
+  
+    guard let email = self.emailField.text, let password = self.passwordField.text else {
+          self.showMessagePrompt("email/password can't be empty")
+      return
+    }
+    
+    showSpinner{}
```

Comment:
Be consistent with `showSpinner` and `self.showSpinner`. Also, please add a space between the method name and the curly braces so it's consistent with the rest of the file.

```swift
self.showSpinner {}
```